### PR TITLE
Add some casts for %j

### PR DESCRIPTION
**Pull request by richsalz**
Let's see how the CI goes.  This was found on Darwin.

### CODE REVIEW

File: apps/enc.c [hunk 1]
```diff
@@ -555,8 +555,10 @@ int enc_main(int argc, char **argv)
 
     ret = 0;
     if (verbose) {
-        BIO_printf(bio_err, "bytes read   : %8ju\n", BIO_number_read(in));
-        BIO_printf(bio_err, "bytes written: %8ju\n", BIO_number_written(out));
+        BIO_printf(bio_err, "bytes read   : %8ju\n",
+                  (uintmax_t)BIO_number_read(in));
```
Comment (paulidale) [inline apps/enc.c]: An extra space indent?
There are more in other files.

File: apps/enc.c [hunk 2]
```diff
@@ -555,8 +555,10 @@ int enc_main(int argc, char **argv)
...
+        BIO_printf(bio_err, "bytes written: %8ju\n",
+                  (uintmax_t)BIO_number_written(out));
```
Comment (paulidale) [inline apps/enc.c]: space

File: test/constant_time_test.c [hunk 3]
```diff
@@ -137,12 +137,14 @@ static int test_select_64(uint64_t a, uint64_t b)
 
     if (selected != a) {
         TEST_error("test_select_64 TRUE failed");
-        BIO_printf(bio_err, "a=%jx b=%jx got %jx wanted a\n", a, b, selected);
+        BIO_printf(bio_err, "a=%jx b=%jx got %jx wanted a\n",
+                  (uintmax_t)a, (uintmax_t)b, (uintmax_t)selected);
```
Comment (paulidale) [inline test/constant_time_test.c]: space

File: test/constant_time_test.c [hunk 4]
```diff
@@ -137,12 +137,14 @@ static int test_select_64(uint64_t a, uint64_t b)
...
         return 0;
     }
     selected = constant_time_select_64(CONSTTIME_FALSE_64, a, b);
     if (selected != b) {
-        BIO_printf(bio_err, "a=%jx b=%jx got %jx wanted b\n", a, b, selected);
+        BIO_printf(bio_err, "a=%jx b=%jx got %jx wanted b\n",
+                  (uintmax_t)a, (uintmax_t)b, (uintmax_t)selected);
```
Comment (paulidale) [inline test/constant_time_test.c]: space

**Comment by richsalz** [comment]
fixup commit pushed; please double-check.

**Comment by richsalz** [comment]
thanks!

**Comment by dot-asm** [comment]
I apologize for crashing the party, but this is not going to work universally. Because there systems, where the type in question is not available. Hint, see e_os2.h. Collateral question in the context if one has considered it a bug in compiler warning subsystem. I mean if it's a bug, then why do we work around it? Or rather if we work around this one, do we work around every other one?

**Comment by richsalz** [comment]
Aargh, you're right.  how about ossl_intmax_t from ossltypes.h?
how about a +1 to revert this?

**Comment by kaduk** [comment]
+1 to revert, yes.

**Pull request closed without merging**